### ISSUE TITLE
Please add support for Buffer

### ISSUE THREAD

**Issue by alecthomas** [ReactiveX/RxSwift#133]
I have a Mac application that monitors a set of directories for file changes. I'm pumping these events through Rx (which makes the code very elegant IMO). When some changes occur (such as rebuilding generated sources) it can trigger a very large number of events.

eg.

```
from([(delete, foo.txt), (delete, bar.txt), (create, bar.txt), (create, foo.txt)])
```

 Ideally I would like to be able to buffer and de-duplicate these events so that only one event per file is emitted in a given time window. Strictly speaking, I would like a buffer that is somewhat like throttle, where it would emit the accumulated array when either the size exceeds a limit, or no new element has been received within some timeout.

So basically, I think support for the Buffer operator would mostly solve this problem.

**Comment by kzaher** [comment]
Buffer and window operators are on list for 2.0 :)

**Comment by kzaher** [comment]
@alecthomas ,

I've added `buffer` operator to `develop` branch. So let me know if that solves your problem.

**Comment by alecthomas** [comment]
It's perfect, thanks very much :)
